src/compiler/binder.ts

```diff
@@ -118,11 +118,7 @@ module ts {
         }
 
         function declareSymbol(symbols: SymbolTable, parent: Symbol, node: Declaration, includes: SymbolFlags, excludes: SymbolFlags): Symbol {
-            // Nodes with computed property names will not get symbols, because the type checker
-            // does not make properties for them.
-            if (hasComputedNameButNotSymbol(node)) {
-                return undefined;
-            }
+            Debug.assert(!hasComputedNameButNotSymbol(node));

```

Comment:
We should discuss this; @yuit and I think that the "But" is a problem - it's just "And" but it leads on to something that is contradictory in intuition.

hasDynamicName is not bad for starters